### PR TITLE
[impala] Fix TSessionHandle guid and secret corrupted after save

### PR DESCRIPTION
**Pull request by wing2fly**
### What changes were proposed in this pull request?

- Fix TSessionHandle guid and secret corrupted after save

### How was this patch tested?

- tested in my local py3 dev environment

### CODE REVIEW

File: apps/beeswax/src/beeswax/models.py [hunk 1]
```diff
@@ -471,7 +471,17 @@ class Session(models.Model):
...
-    secret, guid = HiveServerQueryHandle.get_decoded(secret=self.secret, guid=self.guid)
+    secret = self.secret
+    guid = self.guid
+    if sys.version_info[0] > 2:
+      # only for py3, after bytes saved, bytes wrapped in a string object
+      # a new guid/secret before saved, the eval() will throws exception
+      try:
+        secret = eval(secret)
+        guid = eval(guid)
+      except:
```
Comment (romainr) [inline apps/beeswax/src/beeswax/models.py:482]: Nice!

Just could we just add a test on get_handle()?

Then will be easy to quick check if we can do something slightly cleaner than eval()

For the record on previous commit that broke it: 
https://github.com/cloudera/hue/commit/7d41081166d18a9362a13e332082e5ca64ba7d3a#diff-7eda3aa27bde30ed2e353eca90d753d39a764a4e218676124fd091ead7a97883

File: apps/beeswax/src/beeswax/models.py [hunk 2]
```diff
@@ -471,7 +471,17 @@ class Session(models.Model):
...
 
   def get_handle(self):
-    secret, guid = HiveServerQueryHandle.get_decoded(secret=self.secret, guid=self.guid)
+    secret = self.secret
+    guid = self.guid
+    if sys.version_info[0] > 2:
+      # only for py3, after bytes saved, bytes wrapped in a string object
+      # a new guid/secret before saved, the eval() will throws exception
+      try:
+        secret = eval(secret)
```
Comment (romainr) [inline apps/beeswax/src/beeswax/models.py:480]: Did you check the type of secret and even have a few values that way easy to see when it errors/not error? (can reuse them in the test)

**Review by romainr — COMMENTED** [review]
Nice! Just how about adding proper testing?

File: apps/beeswax/src/beeswax/models.py (hunk 2 above)
Comment (wing2fly) [reply apps/beeswax/src/beeswax/models.py]: Added type checking and reuse in unit test

File: apps/beeswax/src/beeswax/models.py (hunk 1 above)
Comment (wing2fly) [reply apps/beeswax/src/beeswax/models.py]: Added unit test for get_handle()

File: apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py [hunk 3]
```diff
@@ -91,10 +92,28 @@ def test_open_session(self):
         session_count + 1,  # +1 as setUp resets the user which deletes cascade the sessions
         Session.objects.filter(owner=self.user, application=self.query_server['server_name']).count()
       )
+
+      session = Session.objects.get_session(self.user, self.query_server['server_name'])
```
Comment (romainr) [inline apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py:96]: Nice! Loading from the DB

File: apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py [hunk 4]
```diff
@@ -64,6 +64,7 @@ def test_open_session(self):
     )
 
     with patch('beeswax.server.hive_server2_lib.thrift_util.get_client') as get_client:
+      original_secret = original_guid = b'1'
```
Comment (romainr) [inline apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py:67]: Nit: maybe replace it by one of your full string? (more realistic on the issue)

File: apps/beeswax/src/beeswax/models.py [hunk 5]
```diff
@@ -475,11 +475,21 @@ class Session(models.Model):
...
 
     handle_id = THandleIdentifier(secret=secret, guid=guid)
     return TSessionHandle(sessionId=handle_id)
 
+  def get_adjusted_guid_secret(self):
+    secret = self.secret
+    guid = self.guid
+    if sys.version_info[0] > 2 and not isinstance(self.secret, bytes) and not isinstance(self.guid, bytes):
+      # only for py3, after bytes saved, bytes wrapped in a string object
+      secret = eval(secret)
```
Comment (romainr) [inline apps/beeswax/src/beeswax/models.py:489]: Curious: why eval() method? (nothing "cleaner")

**Review by romainr — APPROVED** [review]

File: apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py (hunk 4 above)
Comment (wing2fly) [reply apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py:67]: Done

File: apps/beeswax/src/beeswax/models.py (hunk 5 above)
Comment (wing2fly) [reply apps/beeswax/src/beeswax/models.py:489]: changed to ast.literal_eval() for safe eval
https://stackoverflow.com/questions/15197673/using-pythons-eval-vs-ast-literal-eval

**Pull request merged**